lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1417 @@
+            }
+        }
+        catch {
+            # If cherry-pick fails return to previous state.
+            Write-JujuWarning ("Cherry-pick for {0} failed. Reverting to the previous state.Error: $_" -f
+                              @($commit['branch_ref']))
+            Start-ExternalCommand -ScriptBlock {
+                git cherry-pick --abort
+            }
+        }
+    }
+}
+
+
+function Initialize-GitRepositories {
+    Param(
+        [ValidateSet("hyperv", "ovs")]
+        [string]$NetworkType,
+        [string]$BranchName,
+        [string]$BuildFor
+    )
+
+    Write-JujuLog "Cloning the required Git repositories"
+    
+    $cherryPicks = Get-CherryPicksObject
+    $openstackBuild = Join-Path $BUILD_DIR "openstack"
+    if ($NetworkType -eq 'hyperv') {
+        if ($BuildFor -ne "openstack/networking-hyperv") {
+                Initialize-GitRepository "$openstackBuild\networking-hyperv" $NETWORKING_HYPERV_GIT "master" $cherryPicks['networking-hyperv']
+        }
+    }
+
+    if ($BuildFor -ne "openstack/nova") {
+            Initialize-GitRepository "$openstackBuild\nova" $NOVA_GIT $BranchName $cherryPicks['nova']
+    }
+    
+    if ($BuildFor -ne "openstack/neutron") {
+            Initialize-GitRepository "$openstackBuild\neutron" $NEUTRON_GIT $BranchName $cherryPicks['neutron']
+    }
+    
+    if ($BuildFor -ne "openstack/os-win") {
+            Initialize-GitRepository "$openstackBuild\os-win" $OSWIN_GIT "master" $cherryPicks['os-win']
+    }
+}
+
+
+function Initialize-Environment {
+    Param(
+        [string]$BranchName='master',
+        [string]$BuildFor
+    )
+
+    $dirs = @($CONFIG_DIR, $BIN_DIR, $INSTANCES_DIR, $LOG_DIR, $SERVICE_DIR)
+    foreach($dir in $dirs) {
+        if (!(Test-Path $dir)) {
+            Write-JujuLog "Creating $dir folder."
+            mkdir $dir
+        }
+    }
+
+    $mkisofsPath = Join-Path $BIN_DIR "mkisofs.exe"
+    $qemuimgPath = Join-Path $BIN_DIR "qemu-img.exe"
+    if (!(Test-Path $mkisofsPath) -or !(Test-Path $qemuimgPath)) {
+        Write-JujuLog "Extracting OpenStack binaries"
+        $zipPath = Join-Path $FILES_DIR "openstack_bin.zip"
+        Expand-ZipArchive $zipPath $BIN_DIR
+    }
+  
+    $networkType = Get-JujuCharmConfig -Scope 'network-type'
+    Initialize-GitRepositories $networkType $BranchName $BuildFor
+
+    $standardProjects = 'openstack/nova', 'openstack/neutron', 'openstack/os-win'
+    if ($buildFor -in $standardProjects) {
+        $defaultProjects = $standardProjects -notmatch $buildFor
+        foreach ($project in $defaultProjects) {
+            Install-Projects $project
+        }
+        Install-NetworkType
+        Install-Projects $buildFor
+    }
+    else {
+        foreach ($project in $standardProjects) {
+            Install-Projects $project
+        }
+        Install-NetworkType
+        if (($buildFor -ne "none") -and ($buildFor -ne "openstack/networking-hyperv")) {
```

Comment:
There is object `$null` and not string `"none"`, and also the default value for a string parameter will be the empty string. So, rewrite this as follows:
```
if ($buildFor -and ($buildFor -ne "openstack/networking-hyperv")) {
```